### DISCUSSION TITLE
How to sort by "versions"?

### DISCUSSION THREAD

**Discussion by c33s** [mikefarah/yq#1512] [Q&A]
i would like to get the latest version of a yaml structure which contains semver and datever versions.

```
downloads:
    app1:
        versions:
            2021-07-27T06-46-19Z: ~
    app2:
        versions:
            8.0.0: ~
            2.6.9: ~
            10.0.9: ~
            10.99.4: ~
            2.5.1: ~
            1.0.0: ~
            5.0.0: ~
```
i want to implement `latest` as version which should select the latest version out of the unsorted versions. 

my current command sorts the result "wrong"
```
yq '.downloads.[env(app)].versions | keys | sort | reverse' "$downloads_data_path"
- 8.0.0
- 2.6.9
- 10.0.9
- 10.99.4
- 2.5.1
- 1.0.0
- 5.0.0
```

it should be:
```
            10.99.4: ~
            10.0.9: ~
            8.0.0: ~
            5.0.0: ~
            2.6.9: ~
            2.5.1: ~
            1.0.0: ~
```

**Comment by mikefarah** [comment]
yq doesn't know about semantic version strings (yet) - if you can convert the string to a number that's relative in size to the others, then you can use sort...
